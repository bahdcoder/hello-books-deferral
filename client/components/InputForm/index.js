import React from 'react';
import Input from '@components/Input';

const InputForm = props => {
    const { name, labelName, errors, touched } = props

    return (
        <div>
            <div className="flex mt-4 mb-0 flex-wrap sm:flex-no-wrap">
                <label
                    className="font-raleway py-2 sm:w-5/12 w-full px-4 sm:px-0 text-base text-gray-550 text-left "
                    htmlFor={name}
                >
                    {labelName || name}
                </label>
                <Input {...props} />
            </div>
            {errors[name] && touched[name] && (
                <span className="font-raleway py-0 sm:w-5/12 w-full px-4 text-sm sm:px-0 text-red-500 text-center ">
                    {errors[name]}
                </span>
            )}
        </div>
    );
};

export default InputForm;
