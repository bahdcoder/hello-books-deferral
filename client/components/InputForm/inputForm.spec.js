import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jsdom-global/register';
import InputForm from './index';
const errors = {
    test: 'this is a test'
};
const touched = {
    test: 'this is a test'
};

describe('input Form component', () => {
    it('should render the input-form with a span when errors are available', () => {
        const component = shallow(
            <InputForm errors={errors} touched={touched} name="test" />
        );
        expect(component).toMatchSnapshot();
    });

    it('should render the input-form without a span when errors are not available', () => {
        const component = shallow(
            <InputForm errors={{}} touched={{}} name="test" />
        );

        expect(component).toMatchSnapshot();
    });

    it('should render the input-form with a textarea and not an input when inputtype is textarea', () => {
        const component = shallow(
            <InputForm
                errors={{}}
                touched={{}}
                inputtype="textarea"
                name="password"
            />
        );
        expect(component).toMatchSnapshot();
    });
    it('should render the input-form with a select and not an input when inputtype is select', () => {
        const component = shallow(
            <InputForm
                errors={{}}
                touched={{}}
                inputtype="select"
                name="password"
            />
        );
        expect(component).toMatchSnapshot();
    });
    it('should render the input and label in flex col when block props is true', () => {
        const component = shallow(
            <InputForm
                block={'true'}
                errors={{}}
                touched={{}}
                inputtype="select"
                name="password"
            />
        );
        expect(component).toMatchSnapshot();
    });
    it('testing mount', () => {
        const component = mount(
            <InputForm
                errors={{}}
                touched={{}}
                errors={errors}
                handleChange={handleChange}
                handleBlur={handleBlur}
                name="password"
            />
        );
        expect(component).toMatchSnapshot();
    });
});
