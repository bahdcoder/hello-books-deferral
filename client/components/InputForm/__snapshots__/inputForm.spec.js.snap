// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input Form component should render the input and label in flex col when block props is true 1`] = `
<div
  className="flex mt-4 mb-0 flex-wrap sm:flex-no-wrap flex-col"
>
  <label
    className="font-raleway py-2  w-full px-4 mr-4 sm:px-0 text-base text-gray-550 text-left "
    htmlFor="password"
  >
    password
  </label>
  <div
    className="flex flex-col w-full "
  >
    <Select
      block="true"
      errors={Object {}}
      inputtype="select"
      name="password"
      touched={Object {}}
    />
  </div>
</div>
`;

exports[`input Form component should render the input-form with a select and not an input when inputtype is select 1`] = `
<div
  className="flex mt-4 mb-0 flex-wrap sm:flex-no-wrap "
>
  <label
    className="font-raleway py-2  w-full px-4 mr-4 sm:px-0 text-base text-gray-550 text-left sm:w-4/12"
    htmlFor="password"
  >
    password
  </label>
  <div
    className="flex flex-col w-full sm:w-8/12"
  >
    <Select
      errors={Object {}}
      inputtype="select"
      name="password"
      touched={Object {}}
    />
  </div>
</div>
`;

exports[`input Form component should render the input-form with a span when errors are available 1`] = `
<div
  className="flex mt-4 mb-0 flex-wrap sm:flex-no-wrap "
>
  <label
    className="font-raleway py-2  w-full px-4 mr-4 sm:px-0 text-base text-gray-550 text-left sm:w-4/12"
    htmlFor="test"
  >
    test
  </label>
  <div
    className="flex flex-col w-full sm:w-8/12"
  >
    <Input
      errors={
        Object {
          "test": "this is a test",
        }
      }
      name="test"
      touched={
        Object {
          "test": "this is a test",
        }
      }
    />
    <span
      className="font-raleway py-0 w-full px-4 text-sm sm:px-0 text-red-600 text-left pt-1"
    >
      this is a test
    </span>
  </div>
</div>
`;

exports[`input Form component should render the input-form with a textarea and not an input when inputtype is textarea 1`] = `
<div
  className="flex mt-4 mb-0 flex-wrap sm:flex-no-wrap "
>
  <label
    className="font-raleway py-2  w-full px-4 mr-4 sm:px-0 text-base text-gray-550 text-left sm:w-4/12"
    htmlFor="password"
  >
    password
  </label>
  <div
    className="flex flex-col w-full sm:w-8/12"
  >
    <Textarea
      errors={Object {}}
      inputtype="textarea"
      name="password"
      touched={Object {}}
    />
  </div>
</div>
`;

exports[`input Form component should render the input-form without a span when errors are not available 1`] = `
<div
  className="flex mt-4 mb-0 flex-wrap sm:flex-no-wrap "
>
  <label
    className="font-raleway py-2  w-full px-4 mr-4 sm:px-0 text-base text-gray-550 text-left sm:w-4/12"
    htmlFor="test"
  >
    test
  </label>
  <div
    className="flex flex-col w-full sm:w-8/12"
  >
    <Input
      errors={Object {}}
      name="test"
      touched={Object {}}
    />
  </div>
</div>
`;

exports[`input Form component testing mount 1`] = `
<InputForm
  errors={Object {}}
  name="password"
  touched={Object {}}
>
  <div
    className="flex mt-4 mb-0 flex-wrap sm:flex-no-wrap "
  >
    <label
      className="font-raleway py-2  w-full px-4 mr-4 sm:px-0 text-base text-gray-550 text-left sm:w-4/12"
      htmlFor="password"
    >
      password
    </label>
    <div
      className="flex flex-col w-full sm:w-8/12"
    >
      <Input
        errors={Object {}}
        name="password"
        touched={Object {}}
      >
        <input
          className="font-robotoMono outline-none h-8 text-base px-4 text-gray-550 bg-gray-250 w-full rounded-10"
          errors={Object {}}
          name="password"
          touched={Object {}}
        />
      </Input>
    </div>
  </div>
</InputForm>
`;
